lib/modifiers/flatten.js

```diff
@@ -0,0 +1,60 @@
+var _ = require('lodash');
+var through = require('through2');
+
+/**
+ * Returns a new stream that flattens all arrays passing through by one level. All non-array
+ * items will be passed through as-is.
+ *
+ * @method
+ * @static
+ * @since    1.5.0
+ * @category Util
+ *
+ * @returns {Stream} - Transform stream.
+ *
+ * @example
+ * var input = through.obj();
+ *
+ * input.pipe(nodeStream.flatten());
+ *
+ * input.write([1, 2, 3]);
+ * input.write([4, 5]);
+ * input.write(6);
+ *
+ * // => [1, 2, 3, 4, 5, 6]
+ * @example
+ * var input = through.obj();
+ *
+ * input
+ *   // batch items that are read
+ *   .pipe(nodeStream.batch({ count: 2 }))
+ *   // perform a transform action on the batches
+ *   .pipe(transformBatch())
+ *   // flatten the batches back
+ *   .pipe(nodeStream.flatten());
+ *
+ * input.write(1);
+ * input.write(2);
+ * input.write(3);
+ * input.write(4);
+ * input.write(5);
+ *
+ * // => [1, 2, 3, 4, 5]
+ */
+function flatten() {
+  return through.obj(function Transform(data, enc, cb) {
+    var self = this;
+
+    if (!_.isArray(data)) {
+      return cb(null, data);
+    }
+
+    data.forEach(function (val) {
+      self.push(val);
```

Comment:
So, I'm okay with the rest of the PR and am awaiting results from the back pressure investigation before merging.

It seems like there is certainly an issue since we're expanding a potentially large array. But, that would certainly complicate the implementation since you would have to listen to the back pressure related events. It does seem like a worthwhile venture though and may even be something we can write in a util to be used elsewhere like the parse/stringify utils.

I don't think we have a backpressure issue in any of the other methods currently so if we can write this one with proper backpressure, that would be nice and mean we aren't introducing potentially bad stuff into the code.